### PR TITLE
8327989: java/net/httpclient/ManyRequest.java should not use "localhost" in URIs

### PR DESCRIPTION
**Pull request by dfuch**
Please find here a trivial test fix that removes usage of "localhost" from ManyRequest.java.
Use of "localhost" in test URIs is problematic as it leaves the test open to random failure when running on machines that have custom mappings for "localhost". The fix is to use the IP literal of the loopback address instead.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8327989](https://bugs.openjdk.org/browse/JDK-8327989): java/net/httpclient/ManyRequest.java should not use "localhost" in URIs (**Sub-task** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [9b82022d](https://git.openjdk.org/jdk/pull/18239/files/9b82022ded6f42c3953955a3db5e2c4ca27945f0)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**)
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18239/head:pull/18239` \
`$ git checkout pull/18239`

Update a local copy of the PR: \
`$ git checkout pull/18239` \
`$ git pull https://git.openjdk.org/jdk.git pull/18239/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18239`

View PR using the GUI difftool: \
`$ git pr show -t 18239`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18239.diff">https://git.openjdk.org/jdk/pull/18239.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18239#issuecomment-1991915802)

### CODE REVIEW

**Review by jaikiran — APPROVED** [review]
The change looks good to me. The copyright year on the file would need an update before integrating.

**Comment by dfuch** [comment]
> The change looks good to me. The copyright year on the file would need an update before integrating.

Thanks Jaikiran. Done.

**Review by jaikiran — APPROVED** [review]

File: test/jdk/java/net/httpclient/ManyRequests.java [hunk 1]
```diff
@@ -187,7 +187,7 @@ static void test(HttpsServer server, HttpClient client) throws Exception {
 
         URI baseURI = URIBuilder.newBuilder()
                 .scheme("https")
-                .host(InetAddress.getLoopbackAddress().getHostName())
+                .host(InetAddress.getLoopbackAddress())
```
Comment (djelinski) [inline test/jdk/java/net/httpclient/ManyRequests.java:190]:
```suggestion
                .loopback()
```

**Review by djelinski — APPROVED** [review]

**Review by lgxbslgx — APPROVED** [review]

**Pull request closed without merging**